### PR TITLE
Fix `TraceContextPropagator` behavior when trace parent flags contain…

### PR DESCRIPTION
**Pull request by Kielek**
Fixes test_traceparent_trace_flags_illegal_characters from w3c test suite
Design discussion issue #

## Changes

Fix `TraceContextPropagator` behavior when trace parent flags contains illegal characters.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

**Comment by Kielek** [comment]
@open-telemetry/dotnet-approvers, I have 2 more tests fixes ready to review on separate branch. As it touches some internal implementation/decision it is worth to make separate reviews. If you think otherwise I can bring all of them in one PR.

2 other fixes on the top of branch: https://github.com/Kielek/opentelemetry-dotnet/commits/tace-context-propagator-other-fixes

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
   trace was running (`Activity.Current != null`).
   ([#4890](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4890))
 
+* Fix `TraceContextPropagator` behavior when trace parent flags contains
```
Comment (utpilla) [inline src/OpenTelemetry.Api/CHANGELOG.md:9]: You could rephrase this to something like this:
```md
* Fix `TraceContextPropagator` by validating the first digit of the hex-encoded `trace-flags` field of the `traceparent` header.
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Api/CHANGELOG.md (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Api/CHANGELOG.md]: Fixed: d0df3dbb0ffd83b08def71a65dd95c96cc26e0bf

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs [hunk 2]
```diff
@@ -195,19 +195,20 @@ internal static bool TryExtractTraceparent(string traceparent, out ActivityTrace
...
-        byte options1;
+        byte optionsLowByte;
         try
         {
             spanId = ActivitySpanId.CreateFromString(traceparent.AsSpan().Slice(VersionAndTraceIdLength, SpanIdLength));
-            options1 = HexCharToByte(traceparent[VersionAndTraceIdAndSpanIdLength + 1]);
+            _ = HexCharToByte(traceparent[VersionAndTraceIdAndSpanIdLength]); // to verify if there is no bad chars on options position
+            optionsLowByte = HexCharToByte(traceparent[VersionAndTraceIdAndSpanIdLength + 1]);
         }
         catch (ArgumentOutOfRangeException)
```
Comment (utpilla) [inline src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs:205]: This is outside the scope of this PR but we don't use the exception in any way. We could simply update `HexCharToByte` method to return `false` instead of throwing an unused exception.

Comment (utpilla) [reply src/OpenTelemetry.Api/Context/Propagation/TraceContextPropagator.cs:205]: @Kielek Would you mind updating this follow-up PRs?
